### DISCUSSION TITLE
Issue about Numpy 2.x vs 1.x

### DISCUSSION THREAD

**Discussion by none-of-my-names-is-available** [zwicker-group/py-pde#575] [Q&A]
Hi,

having installed py-pde, I tried the first example in the documentation involving a Diffusion PDE on a 64 by 64 grid.
on "import py-pde", I get the message:

_A module that was compiled using NumPy 1.x cannot be run in
NumPy 2.0.0 as it may crash. To support both 1.x and 2.x
versions of NumPy, modules must be compiled with NumPy 2.0.
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.

If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2._

I did some more simple things with python during the last two years, and I never got a message like this. Does anyone know, how to "compile the module" (py-pde?) for numpy 2.0? I thought that python is an interpreter.

Many thanks in advance

Günter Becker

**Comment by david-zwicker** [comment]
`py-pde` supports Numpy 2.0, so it seems as if another module is the culprit. Please update all other packages (in particular `numba`, `scipy`, and `matplotlib`) to see whether they cause the issue.

**Reply by none-of-my-names-is-available** [reply]
Thank you for the reply.
My first solution has been:
_pip uninstall numpy
pip install numpy==1.26_
with 1.26 the last version of numpy before 2.0
According to your suggestion, I did:
_pip uninstall matplotlib
pip install matplotlib_
and went back to numpy 2.0

This also worked, and it keeps me up to date with current numpy.

The other packages needed no change.

Best regards

Günter

**Comment by denis-bz** [comment]
A comment today (8 Nov) on https://github.com/numba/numba/issues/9708
> Any word on whether we can expect a numba 0.61 release pinning numpy to 2.1 any time soon ?

(+++ py-pde looks *very* nice, to a pde noob)
